### PR TITLE
Contributing information and application

### PR DESCRIPTION
**Pull request by EvilGiraffes**
This pull request adds the contributing file and changes the current code to fit

It also adds some contribution help and explains the process of how to run the code to be able to contribute

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+## Style
+
+| Kind                      | Case          |
+|:--------------------------|:--------------|
+| **Class**                 | ClassName     |
+| **Private Class**         | PClassName    |
+| **Class Fields**          | fieldName     |
+| **Class Methods**         | methodName    |
+| **Private Class Fields**  | pfieldName    |
+| **Private Class Methods** | pmathodName   |
```
Comment (EvilGiraffes) [inline CONTRIBUTING.md]: rah typo forgot and forgot to uppercase

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Style
+
+| Kind                      | Case          |
+|:--------------------------|:--------------|
+| **Class**                 | ClassName     |
+| **Private Class**         | PClassName    |
+| **Class Fields**          | fieldName     |
+| **Class Methods**         | methodName    |
+| **Private Class Fields**  | pfieldName    |
```
Comment (EvilGiraffes) [inline CONTRIBUTING.md]: also forgot to uppercase here

File: src/tools.py [hunk 3]
```diff
@@ -121,7 +121,8 @@ def _colorsToHex(r: int = 0, g: int = 0, b: int = 0) -> str:
     
     return str.lower(f"#{r:02X}{g:02X}{b:02X}")
 
-def _formatForSVG(char: str):
-    if char in SVGMaps:
-        char = SVGMaps.get(char)
-    return char
+def pFormatForSVG(char: str) -> str | None:
```
Comment (AndreasTar) [inline src/tools.py]: this is wrong. The original function returns the input character if it is not inside the SVG_MAPS list, whereas this version would return None, making the `converters.ConvertToSVG()` output wrong results in line 42 of the file src/converters.py. This function should never return none.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+### Roadmap
+
+The roadmap can be found in the top level readme file of the project, this includes a todo list aswell.
+
+### Todo list
+
+There are currently two todo lists:
+
+1. top level readme under Roadmap
+2. src/readme under TODOs
```
Comment (AndreasTar) [inline CONTRIBUTING.md]: Also add the various small TODOs and FIXMEs etc scattered throughout the codebase.

**Review by AndreasTar — CHANGES_REQUESTED** [review]

File: src/tools.py (hunk 3 above)
Comment (EvilGiraffes) [reply src/tools.py]: the original function would return None as it was, but i will change it to ensure it returns char back as a default
Edit: i was wrong

**Review by AndreasTar — APPROVED** [review]

File: src/frontend.py [hunk 5]
```diff
@@ -245,39 +246,40 @@ def RunTool(shouldSave: bool = False): # TODO implement shouldSave, if false ret
...
             return userin
         
-def Exit(error: str = None):
+def exitWith(error: Exception | str | None = None) -> NoReturn:
     if error:
         print(f"\nDuring processing, program encountered an error with the message:\n\t{error}\n")
     print("\nTool is exiting...\n")
-    exit()
+    # Non zero exit code indicates something went wrong
+    exit(1)
```
Comment (EvilGiraffes) [inline src/frontend.py]: should have an exit code set when the if condition is true

**Comment by EvilGiraffes** [comment]
I have made quite a change to the main readme now, hopefully this is okay

File: README.md [hunk 6]
```diff
@@ -1,13 +1,49 @@
...
   
  This source code is licensed under the BSD-style license found in the **LICENSE** file in the root directory of this Repository.
 
-## Usage
+## How to run
+
+### Environment
+
+**If you have the packages in [requirements.txt](requirements.txt) installed on your root environment 
+or wish to install it on your root environmen you can skip this section**.
```
Comment (AndreasTar) [inline README.md:17]: typo

**Review by AndreasTar — APPROVED** [review]

**Review by AndreasTar — APPROVED** [review]

**Pull request merged**